packages/itwinui-react/src/core/Menu/MenuItem.tsx

```diff
@@ -224,107 +145,70 @@ export const MenuItem = React.forwardRef((props, forwardedRef) => {
     // Focus the item when hovered.
     if (e.target === e.currentTarget) {
       menuItemRef.current?.focus();
-
-      // Since we manually focus the MenuItem on hover, we need to manually update the active index for
-      // Floating UI's keyboard navigation to work correctly.
-      if (parent != null && focusableNodeIndexInParentTree != null) {
-        parent.setCurrentFocusedNodeIndex(focusableNodeIndexInParentTree);
-      }
     }
   };
 
-  const onFocus = () => {
-    // Set hasFocusedNodeInSubmenu in a microtask to ensure the submenu stays open reliably.
-    // E.g. Even when hovering into it rapidly.
-    queueMicrotask(() => {
-      parent.setHasFocusedNodeInSubmenu?.(true);
-    });
-
-    tree?.events.emit('onNodeFocused', {
-      nodeId,
-      parentId,
-    } satisfies TreeEvent);
-  };
-
   const onClick = () => {
     onClickProp?.(value);
-    setIsSubmenuVisible((prev) => !prev);
   };
 
   const handlers = !disabled
     ? {
         onClick,
         onKeyDown,
         onMouseEnter,
-        onFocus,
       }
     : {};
 
+  const trigger = (
+    <ListItem
+      as='div'
```

Comment:
Should we use `button` so that it automatically handles Enter/Spacebar keys?

Reply:
> Should we use `button` so that it automatically handles Enter/Spacebar keys?

I tried using `as="button"`, but the styles broke:

<details>
<summary>Screenshot</summary>

<img width="291" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/0fb9de2d-67d5-4c81-9431-b4638f8ac498">

</details>

> There's also `ListItem.Action` which renders a button.

I tried wrapping the `children` with a `ListItem.Action` but for some reason, could not get `ListItem.Action`'s `onClick` to trigger when pressing <kbd>Enter</kbd>/<kbd>Space</kbd>.

<details>
<summary>diff</summary>

```diff
--- a/packages/itwinui-react/src/core/Menu/MenuItem.tsx
+++ b/packages/itwinui-react/src/core/Menu/MenuItem.tsx
@@ -123,24 +123,6 @@ export const MenuItem = React.forwardRef((props, forwardedRef) => {
     } satisfies Parameters<typeof Menu>[0]['popoverProps'];
   }, [subMenuItems.length]);
 
-  const onKeyDown = (event: React.KeyboardEvent<HTMLElement>) => {
-    if (event.altKey) {
-      return;
-    }
-
-    switch (event.key) {
-      case 'Enter':
-      case ' ':
-      case 'Spacebar': {
-        onClick();
-        event.preventDefault();
-        break;
-      }
-      default:
-        break;
-    }
-  };
-
   const onMouseEnter = (e: React.MouseEvent<HTMLElement>) => {
     // Focus the item when hovered.
     if (e.target === e.currentTarget) {
@@ -148,14 +130,8 @@ export const MenuItem = React.forwardRef((props, forwardedRef) => {
     }
   };
 
-  const onClick = () => {
-    onClickProp?.(value);
-  };
-
   const handlers = !disabled
     ? {
-        onClick,
-        onKeyDown,
         onMouseEnter,
       }
     : {};
@@ -185,7 +161,14 @@ export const MenuItem = React.forwardRef((props, forwardedRef) => {
         </ListItem.Icon>
       )}
       <ListItem.Content>
-        <div>{children}</div>
+        <ListItem.Action
+          onClick={() => {
+            console.log('Testing the onClick event.');
+            onClickProp?.(value);
+          }}
+        >
+          {children}
+        </ListItem.Action>
         {sublabel && <ListItem.Description>{sublabel}</ListItem.Description>}
       </ListItem.Content>
       {!endIcon && subMenuItems.length > 0 && (
```

</details>

Reply:
> I tried using `as="button"`, but the styles broke

Yeah that's expected, because `button` comes with a bunch of default styles. The styles are fixable using something like `class=iui-button-base`.

> I tried wrapping the `children` with a `ListItem.Action` but for some reason, could not get `ListItem.Action`'s `onClick` to trigger when pressing <kbd>Enter</kbd>/<kbd>Space</kbd>.

Yeah I realized `ListItem.Action` might not be the best way to do it. It's designed for simple scenarios where a regular non-interactive list item contains an anchor or button. But in this case we are turning `ListItem` into an interactive `option` which gets focused. (Events only fire on the node that gets focus, which in this case is a parent of `ListItem.Action`)

Reply:
> Would this change in types be allowed in a `minor` release? E.g. the event's type in the `onClick` might cause some errors if the users' code

We can leave the types unchanged (as `'div'` rather than `'button'`), it's not super important. Think of `button` as an implementation detail that users should not have to think about.

Since the plan is also to deprecate `MenuItem` in the future, we can have correct types for the replacement components.

Reply:
> In [`b8627fd`](https://github.com/iTwin/iTwinUI/pull/2021/commits/b8627fd8fe29045b7b55e02ce4653ddafeced97c), I also had to make some changes in the unit tests to have the keyboard event trigger the `onClick`. ([Relevant issue](https://redirect.github.com/testing-library/user-event/issues/179))

I'm not sure that test is even necessary anymore, it's just testing the default behavior of buttons. We can remove it, as long as we still have larger tests where we test arrow keys for cycling focus and <kbd>Enter</kbd> key for selection.